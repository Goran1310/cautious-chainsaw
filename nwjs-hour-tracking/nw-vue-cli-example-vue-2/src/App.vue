<template>
  <div id="app">
    <img
      src="@/assets/logo.png"
      alt="Vue.js logo"
      title="Vue.js"
      class="logo"
    />
    <img
      v-if="isDesktop"
      src="@/assets/nw.png"
      alt="NW.js logo"
      title="NW.js"
      class="logo"
    />
    <HelloWorld :msg="message" />
  </div>
</template>

<script>
import HelloWorld from '@/components/HelloWorld.vue';

export default {
  name: 'App',
  components: {
    HelloWorld
  },
  computed: {
    message: function () {
      if (this.isDesktop) {
        return 'Welcome to your Vue.js Desktop App in NW.js!';
      }
      return 'Welcome to your Vue.js Web App!';
    }
  }
};
</script>

<style>
#app {
  margin-top: 60px;
  color: #2C3E50;
  font-family: 'Avenir', sans-serif;
  text-align: center;
}
.logo {
  max-height: 140px;
  margin: 0px 10px;
}
</style>
