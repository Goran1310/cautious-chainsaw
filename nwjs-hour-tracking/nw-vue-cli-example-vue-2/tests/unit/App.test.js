import { shallowMount } from '@vue/test-utils';

import App from '@/App.vue';

describe('App.vue', () => {
  describe('Desktop', () => {
    test('Render default contents', () => {
      const wrapper = shallowMount(App);

      expect(wrapper)
        .toMatchSnapshot();
    });
  });

  describe('Web', () => {
    beforeEach(() => {
      window.webSetup();
    });

    test('Render default contents', () => {
      const wrapper = shallowMount(App);

      expect(wrapper)
        .toMatchSnapshot();
    });
  });
});
