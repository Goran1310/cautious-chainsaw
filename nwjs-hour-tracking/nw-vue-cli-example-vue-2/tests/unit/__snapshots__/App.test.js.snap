// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Desktop Render default contents 1`] = `
<div id="app">
  <img alt="Vue.js logo" class="logo" src="@/assets/logo.png" title="Vue.js">
  <img alt="NW.js logo" class="logo" src="@/assets/nw.png" title="NW.js">
  <helloworld-stub msg="Welcome to your Vue.js Desktop App in NW.js!"></helloworld-stub>
</div>
`;

exports[`App.vue Web Render default contents 1`] = `
<div id="app">
  <img alt="Vue.js logo" class="logo" src="@/assets/logo.png" title="Vue.js">
  <!---->
  <helloworld-stub msg="Welcome to your Vue.js Web App!"></helloworld-stub>
</div>
`;
